app/code/Magento/CatalogStorefront/Model/VariantService.php

```diff
@@ -0,0 +1,283 @@
+     * @var DeleteVariantsResponseFactory
+     */
+    private $deleteVariantsResponseFactory;
+
+    /**
+     * @var ProductVariantsDataProvider
+     */
+    private $productVariantsDataProvider;
+
+    /**
+     * @var ProductVariantMapper
+     */
+    private $productVariantMapper;
+
+    /**
+     * @var CatalogService
+     */
+    private $catalogService;
+
+    /**
+     * @var ProductsGetRequestInterfaceFactory
+     */
+    private $productsGetRequestInterfaceFactory;
+
+    /**
+     * @param ImportVariantsResponseFactory $importVariantsResponseFactory
+     * @param DeleteVariantsResponseFactory $deleteVariantsResponseFactory
+     * @param LoggerInterface $logger
+     * @param CatalogRepository $catalogRepository
+     * @param ProductVariantsDataProvider $productVariantsDataProvider
+     * @param ProductVariantMapper $productVariantMapper
+     * @param CatalogService $catalogService
+     * @param ProductsGetRequestInterfaceFactory $productsGetRequestInterfaceFactory
+     */
+    public function __construct(
+        ImportVariantsResponseFactory $importVariantsResponseFactory,
+        DeleteVariantsResponseFactory $deleteVariantsResponseFactory,
+        LoggerInterface $logger,
+        CatalogRepository $catalogRepository,
+        ProductVariantsDataProvider $productVariantsDataProvider,
+        ProductVariantMapper $productVariantMapper,
+        CatalogService $catalogService,
+        ProductsGetRequestInterfaceFactory $productsGetRequestInterfaceFactory
+    ) {
+        $this->importVariantsResponseFactory = $importVariantsResponseFactory;
+        $this->logger = $logger;
+        $this->catalogRepository = $catalogRepository;
+        $this->deleteVariantsResponseFactory = $deleteVariantsResponseFactory;
+        $this->productVariantsDataProvider = $productVariantsDataProvider;
+        $this->productVariantMapper = $productVariantMapper;
+        $this->catalogService = $catalogService;
+        $this->productsGetRequestInterfaceFactory = $productsGetRequestInterfaceFactory;
+    }
+
+    /**
+     * Import requested variants in storage
+     *
+     * @param ImportVariantsRequestInterface $request
+     * @return ImportVariantsResponseInterface
+     */
+    public function importProductVariants(ImportVariantsRequestInterface $request): ImportVariantsResponseInterface
+    {
+        try {
+            $variantsInElasticFormat = [];
+            foreach ($request->getVariants() as $variantData) {
+                $optionValues = $variantData->getOptionValues();
+                $id = $variantData->getId();
+                $explodedId = explode('/', $id);
+                $parentId = $explodedId[1];
+                $childId = $explodedId[2];
+                foreach ($optionValues as $optionValue) {
+                    preg_match('/(?<=:)(.*)(?=\/)/', $optionValue, $match);
+                    $attrCode = $match[0];
+                    $variant = [
+                        '_id' => $id . '/' . $attrCode,
```

Comment:
why do we need custom value with attrCode for system "_id"?

Reply:
The variant is split up by option_values, and thus the variant_id is not unique anymore and would not work for _id